module/zfs/dmu_tx.c

```diff
@@ -1107,16 +1127,16 @@ dmu_tx_assign(dmu_tx_t *tx, uint64_t txg_how)
 		if (err != ERESTART || !(txg_how & TXG_WAIT))
 			return (err);
 
-		dmu_tx_wait(tx);
+		dmu_tx_wait_flags(tx, txg_how);
```

Comment:
Per [previous comment](https://github.com/openzfs/zfs/pull/11082#discussion_r1160673078), I think this should be:

```suggestion
		dmu_tx_wait_flags(tx,
		    (txg_how & TXG_NOSUSPEND) ? TXG_WAIT_F_NOSUSPEND : 0);
```

I think things mostly work right without this change, but only by accident:
* `TXG_WAIT` becomes `TXG_WAIT_F_SIGNAL`, so it stills waits for the txg to be committed. If it got interrupted somehow, it'd end up back in `dmu_tx_try_assign` and would just go around again
* `TXG_NOTHROTTLE` becomes `TXG_WAIT_F_NOSUSPEND`, which almost certainly does nothing since we just checked for suspend in `dmu_tx_try_assign` and will again in `dmu_tx_wait_flags`, and if it does make it down to `txg_wait_synced_tx` and manages to hit a suspend thers, again, it'll return and end up in `dmu_tx_try_assign` again
* `TXG_NOSUSPEND` becomes `TXG_WAIT_F_FORCED_EXPORT`, which is really just a suspend check with a different error code, but since we don't check the error code, again, we end up in `dmu_tx_try_assign`

I think this is the smallest possible change that makes it clear.

Reply:
Thank you applied.